cmd/whitesourceExecuteScan.go

```diff
@@ -0,0 +1,370 @@
+package cmd
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+
+	"github.com/SAP/jenkins-library/pkg/command"
+	"github.com/SAP/jenkins-library/pkg/log"
+	"github.com/SAP/jenkins-library/pkg/telemetry"
+	"github.com/SAP/jenkins-library/pkg/whitesource"
+	"golang.org/x/mod/modfile"
+)
+
+func whitesourceExecuteScan(config whitesourceExecuteScanOptions, telemetryData *telemetry.CustomData) {
+	// for command execution use Command
+	c := command.Command{}
+	sys := whitesource.NewSystem(config.ServiceURL, config.OrgToken, config.UserToken)
+	// reroute command output to logging framework
+	c.Stdout(log.Writer())
+	c.Stderr(log.Writer())
+
+	// error situations should stop execution through log.Entry().Fatal() call which leads to an os.Exit(1) in the end
+	err := runWhitesourceExecuteScan(config, sys, telemetryData, &c)
+	if err != nil {
+		log.Entry().WithError(err).Fatal("step execution failed")
+	}
+}
+
+func runWhitesourceExecuteScan(config whitesourceExecuteScanOptions, sys whitesource.System, telemetryData *telemetry.CustomData, command *command.Command) error {
```

Comment:
unused parameter (for now)
```suggestion
func runWhitesourceExecuteScan(config whitesourceExecuteScanOptions, sys whitesource.System, _ *telemetry.CustomData, command *command.Command) error {
```